src/testutil/message_hexdump.go

```diff
@@ -0,0 +1,111 @@
+package testutil
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/skycoin/skycoin/src/cipher"
+	"github.com/skycoin/skycoin/src/cipher/encoder"
+	"github.com/skycoin/skycoin/src/daemon/gnet"
+)
+
+func serializeMessage(msg gnet.Message) []byte {
```

Comment:
this is a copy of `encodeMessage` from gnet, should make `encodeMessage` a public method and use that

Reply:
#970 exports `gnet.EncodeMessage`

Reply:
Very nice addition . This is something I needed for #735 and even implemented in 42fd42cf .

Reply:
It is a copy, but when I try to use `gnet.EncodeMessage`, I get this error:

`panic: Attempted to serialize message struct not in MessageIdMap: &{1234 7890 5 <nil> false}`

That's why I remove the subsequent error verification in `serializeMessage`. But removing it in the original would spur further errors.

Reply:
Also, there is still the problem of casting `PingMessage` and `PongMessage` as `gnet.Message` type

Reply:
`panic: Attempted to serialize message struct not in MessageIdMap: &{1234 7890 5 <nil> false}` is the message not registered?

Reply:
Problem solved